internal/k8s_watcher/watcher.go

```diff
@@ -98,26 +94,28 @@ func WithBlocktxTicker(t Ticker) func(*Watcher) {
 }
 
 func (c *Watcher) Start() error {
-	c.watchMetamorph()
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Maybe creation of context, it's cancel function and incrementing waitgroup counter could be moved to the methods that use them?
Like  in `internal/metamorph/processor.go line 215` from PR #432

Reply:
Basically that wouldn't be the correct style. We pass the context to the function from outside to pass cancellation and whatever data we have. If we create those ctx and cancel function inside it won't make sense. I'm fine having those inside but just doesn't seem right.